### PR TITLE
fix(angular): provide constructor dependencies

### LINKED ISSUE #21051: [Bug]: Angular: ControlValueAccessor is always set to null
### Describe the bug

When we create an angular component with ControlValueAccessor, we can't change the value of the control. He always seems to be reset to null. 

The behavior don't appear with the alpha version of storybook 7. 

### To Reproduce

```shell
https://github.com/jadeen/storybook-ngcontrol

there is a test component and the story can be found inside the file button.stories.ts
```

### System

```shell
Environment Info:

  System:
    OS: Linux 5.15 Ubuntu 22.04.1 LTS 22.04.1 LTS (Jammy Jellyfish)
    CPU: (8) x64 Intel(R) Core(TM) i7-8550U CPU @ 1.80GHz
  Binaries:
    Node: 16.18.1 - ~/.nvm/versions/node/v16.18.1/bin/node
    Yarn: 1.22.19 - /usr/bin/yarn
    npm: 8.19.2 - ~/.nvm/versions/node/v16.18.1/bin/npm
  Browsers:
    Chrome: 110.0.5481.77
    Firefox: 109.0.1
  npmPackages:
    @storybook/addon-essentials: ^7.0.0-beta.46 => 7.0.0-beta.46 
    @storybook/addon-interactions: ^7.0.0-beta.46 => 7.0.0-beta.46 
    @storybook/addon-links: ^7.0.0-beta.46 => 7.0.0-beta.46 
    @storybook/angular: ^7.0.0-beta.46 => 7.0.0-beta.46 
    @storybook/blocks: ^7.0.0-beta.46 => 7.0.0-beta.46 
    @storybook/testing-library: ^0.0.14-next.1 => 0.0.14-next.1 
```

### Additional context

_No response_

### LINKED ISSUE #21052: [Bug]: Angular: No provider for XXX
### Describe the bug

He had the same behavior that this issue with `httpClient` #20821.

We import `ReactiveFormsModule` and we still need to provide `NgControl`. 

### To Reproduce

```shell
https://github.com/jadeen/storybook-ngcontrol

You only need to remove the provider of NgControl inside the story Button to produce the issue.
```

### System

```shell
System:
    OS: Linux 5.15 Ubuntu 22.04.1 LTS 22.04.1 LTS (Jammy Jellyfish)
    CPU: (8) x64 Intel(R) Core(TM) i7-8550U CPU @ 1.80GHz
  Binaries:
    Node: 16.18.1 - ~/.nvm/versions/node/v16.18.1/bin/node
    Yarn: 1.22.19 - /usr/bin/yarn
    npm: 8.19.2 - ~/.nvm/versions/node/v16.18.1/bin/npm
  Browsers:
    Chrome: 110.0.5481.77
    Firefox: 109.0.1
  npmPackages:
    @storybook/addon-essentials: ^7.0.0-beta.46 => 7.0.0-beta.46 
    @storybook/addon-interactions: ^7.0.0-beta.46 => 7.0.0-beta.46 
    @storybook/addon-links: ^7.0.0-beta.46 => 7.0.0-beta.46 
    @storybook/angular: ^7.0.0-beta.46 => 7.0.0-beta.46 
    @storybook/blocks: ^7.0.0-beta.46 => 7.0.0-beta.46 
    @storybook/testing-library: ^0.0.14-next.1 => 0.0.14-next.1
```

### Additional context

_No response_

### LINKED ISSUE #21121: [Bug]: Angular APP_INITIALIZER not firing in 7.0.0-beta.48
### Describe the bug

When using storybook 7.0.0-beta.48 in angular, the APP_INITIALIZER function is not called before the application is loaded. It seems just to be ignored.

```
import { moduleMetadata, StoryObj } from '@storybook/angular';
import { Meta } from '@storybook/angular';
import { APP_INITIALIZER } from '@angular/core';

const meta: Meta<ButtonComponent> = {
    component: ButtonComponent,
    decorators: [
        moduleMetadata({
            providers: [
                {
                    provide: APP_INITIALIZER,
                    useFactory: () => {
                        console.log('never called');
                    },
                    multi: true
                }
            ]
        })
    ]
};
```

### To Reproduce

_No response_

### System

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by sheriffMoose**
Resolves #21052
Resolves #21051
Resolves #21121

## What I did

* [x] provide moduleMetadata.providers into the root injector, still being provided into the standalone component for overrides
* [x] provide dependencies injected into components constructors into the root injector, in case they are providedIn: 'root' but they are missed by the standalone injector

## How to test

* [ ] Unit Tests Pass
* [ ] CI Passes

## Checklist

- [x] Make sure your changes are tested (stories and/or unit, integration, or end-to-end tests)
- [ ] Make sure to add/update documentation regarding your changes
- [ ] If you are deprecating/removing a feature, make sure to update
      [MIGRATION.MD](https://github.com/storybookjs/storybook/blob/next/MIGRATION.md)

#### Maintainers

- [ ] If this PR should be tested against many or all sandboxes,
      make sure to add the `ci:merged` or `ci:daily` GH label to it.
- [x] Make sure this PR contains **one** of the labels below.

`["cleanup", "BREAKING CHANGE", "feature request", "bug", "documentation", "maintenance", "dependencies", "other"]`

### CODE REVIEW

**Comment by valentinpalkovic** [comment]
@sheriffMoose Can I help you with the failing UI Tests or do you know the issue?

**Comment by sheriffMoose** [comment]
@valentinpalkovic yes please, didn't get the time to look at it yet

**Review by valentinpalkovic — APPROVED** [review]

File: code/frameworks/angular/package.json [hunk 1]
```diff
@@ -77,6 +77,7 @@
     "@angular/forms": "^15.1.1",
     "@angular/platform-browser": "^15.1.1",
     "@angular/platform-browser-dynamic": "^15.1.1",
+    "@ngxs/store": "^3.7.6",
```
Comment (nzacca) [inline code/frameworks/angular/package.json:80]: @valentinpalkovic Is this still needed here? I don't see any references to this package below. I assume this was added just for testing?

Comment (valentinpalkovic) [reply code/frameworks/angular/package.json:80]: Thank you! I will remove it again :)

**Pull request merged**